### ISSUE TITLE
[Feature Request] Allow for choosing root directory for host->guest file sharing?

### ISSUE THREAD

**Issue by XanderThunder** [casualsnek/cassowary#29]
Hi, before requesting the feature, I wanted to say that I totally love your program. It makes using a windows VM so much easier and more integrated. Thank you very much for creating this!!!

**Regarding the feature**:
When exploring the settings of cassowary I noticed that the root directory of the host gets shared no matter what. I guess this is needed since xfreerdp only allows for one shared folder (I might be wrong here). With respect to that, I would like to be able to choose which directory should be the "root" for all other shares. I don't like the idea of my windows vm having access to my root filesystem with all the sensitive data stored on there. 
So instead of making "/" the root share for all other mappings above I would like to select /mnt/xxx and then be able to just map folders inside /mnt/xxx/* to drive letters. Would this be possible?
I honestly don't mind errors appearing when trying to map something outside of /mnt/xxx/.

Cheers!

**Comment by casualsnek** [comment]
This might require a lot of changes to do! 

Host root is shared to ensure that any file on your filesystem can be opened with windows application which a lot of people expect to work without messing up with configurations!
I understand why you might not want / to be shared, I will check the code for path translation and see if it is doable without breaking other stuff. If i figure it out i will either share patch files which you can use, or if feasible i will add a option to configure it from GUI on next release !

**Comment by XanderThunder** [comment]
Well, thanks for the lightning fast reply! I didn't know there was other stuff relying on it. Now I understand why you chose to go that way. Thank you very much for looking into it, I'll be happy to assist with testing. 

If you come to the conclusion that selecting another root share breaks a lot of stuff, I also wouldn't mind having an option to disable it completely for now (then I might be able to use the xfreerpd CLI parameters on the first config page to share my folder that way).

**Comment by casualsnek** [comment]
I think it can be done properly, give me some days and i will add a parameter on config file that you can change !
Although you can't open any file outside of your set root with windows applications !

**Comment by casualsnek** [comment]
Hello, i have done some work on it and pushed the code changes. 
Can you install the built files from here: https://ufile.io/f/p5tod

After installing the windows component, you may need to reboot VM, the setting for shared root is on advanced tab.

Waiting for your feedback :)

**Comment by XanderThunder** [comment]
Hi,
thank you sooo much for implementing this so quickly! I tested the main functions of the app in your provided temporary build and nearly everything seemed to work just fine.

Did you intentionally remove the field to provide the folderpath in the "Folder Mappings -> Create New Share" dialog since I said that I was totally fine with it not being there and me doing additional mappings manually instead? If so, then this behaviour is to be expected.

Again, thank you very much for doing this extra work for me and others that might have similar requirements.

**Here are a few screenshots of my configuration and the findings of my testing:**

![image](https://user-images.githubusercontent.com/57769828/153062272-b430460d-4e5a-4209-a855-86796a80d251.png)
![image](https://user-images.githubusercontent.com/57769828/153062765-6c0867bf-8ee3-4b74-b227-b3fbdfb775b7.png)
![image](https://user-images.githubusercontent.com/57769828/153063423-0396bd9a-fa1f-4de0-af76-b0632f26d4c6.png)
![image](https://user-images.githubusercontent.com/57769828/153063438-2f0e487e-3d1a-43f3-84ca-3466cf5f65c2.png)

**Comment by casualsnek** [comment]
> Did you intentionally remove the field to provide the folderpath in the "Folder Mappings -> Create New Share" dialog since I said that I was totally fine with it not being there and me doing additional mappings manually instead? If so, then this behaviour is to be expected.

I guess you are on wrong tab, its for mounting local disk on windows to linux, its mountpoint location is specified on advanced as Windows share mount root

For Linux to windows sharing use linux-> windows sub tab !

**Comment by XanderThunder** [comment]
You are absolutely right! I mixed up the tabs because in the previous version, the "Linux->Windows" tab was the first one. I guess I did not look closely enough. Sorry for that!

The drive mappings work (after applying the workaround from [issue 11](https://github.com/casualsnek/cassowary/issues/11#issuecomment-1030866830)).
If you tried to map a directory outside of the root share directory you would get an error with a helpful explanation - which works perfectly.
And if you afterwards remove a drive mapping in cassowary while Windows is running, the drive becomes inaccessible through explorer. This solves itself by issuing a restart.

**Here are (again) my screenshots**:

![image](https://user-images.githubusercontent.com/57769828/153067507-182aa2a6-f42f-43b2-8a55-a6404a0729ea.png)
![image](https://user-images.githubusercontent.com/57769828/153066971-7c3e9d7a-8cfe-4bcf-8b48-cabf28c96aab.png)
![image](https://user-images.githubusercontent.com/57769828/153067005-6f25f359-7752-49c1-90d8-eb3bebb5add4.png)
